packages/block-editor/src/store/selectors.js

```diff
@@ -2751,7 +2751,7 @@ export function wasBlockJustInserted( state, clientId, source ) {
  * @return {boolean} True if the block is visible.
  */
 export function isBlockVisible( state, clientId ) {
-	return state.blockVisibility?.[ clientId ] ?? true;
```

Comment:
This was added in https://github.com/WordPress/gutenberg/pull/41104 and was always defaulted to true.